source/MaterialXGenShader/Nodes/SourceCodeNode.cpp

```diff
@@ -81,13 +81,44 @@ void SourceCodeNode::emitFunctionCall(const ShaderNode& node, GenContext& contex
         static const string prefix("{{");
         static const string postfix("}}");
 
-        // Inline expressions can only have a single output
+        size_t pos = 0;
+        size_t i = _functionSource.find_first_of(prefix);
+        int variableIndex = 1;
+        while (i != string::npos)
+        {
+            size_t j = _functionSource.find_first_of(postfix, i + 2);
+            if (j == string::npos)
+            {
+                throw ExceptionShaderGenError("Malformed inline expression in implementation for node " + node.getName());
+            }
+
+            const string variable = _functionSource.substr(i + 2, j - i - 2);
+            const ShaderInput* input = node.getInput(variable);
+            if (!input)
+            {
+                throw ExceptionShaderGenError("Could not find an input named '" + variable +
+                    "' on node '" + node.getName() + "'");
+            }
+
+            if (!input->connection)
+            {
+                string variableName = node.getName() + input->name + std::to_string(variableIndex++);
```

Comment:
Forgot to mention this before but it seems a "_" should be added between the node name and the input for consistency in names.